### PR TITLE
add banner for shabadavali dvn campaign

### PR DESCRIPTION
**Pull request by MySmilingTurban**
* [x]  Passed [sanity tests](http://bit.ly/sttm-sanity-tests).
* [x]  Ran `npm test` & fixed newly introduced lint errors.
* [x]  Checked console for errors.

This PR adds text for promotion of Shabadavali DVN campaign.

Here is a screenshot showing the banner.
image

### CODE REVIEW

**Comment by Gauravjeetsingh** [comment]
@MySmilingTurban veerji, can you share the screenshot of the banner here, I am unable to see the deployments.

**Comment by MySmilingTurban** [comment]
Sure, I am waiting on reply for some design and text changes.
Added the screenshot of the current.

File: src/js/components/Banner/ShabadavaliBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      data: {
+        message: "Help us preserve the Punjabi language! Donate on Dasvandh Network and double your impact with matching!",
+        link: "dvnetwork.org/learnpunjabi",
+        label: "Donate now",
+        type: "3",
+      }
+    }
+  };
+
+  const lastSeen = sessionStorage.getItem("bannerMessage");
```
Comment (gurjit03) [inline src/js/components/Banner/ShabadavaliBanner.tsx:22]: @MySmilingTurban Just wondering we are using the same localStorage item for multiple banners. Suppose a case where we wants to display another error banner due to some outage or so. In this case, if the localStorage item is already set to `seen`, we will miss this `outageBanner`. 

So suggestion will be to keep seperate state for banners. Maybe creating one object banners and having all the banner under that can help us.

File: src/js/components/Banner/ShabadavaliBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import Banner from './Banner';
+
+const ShabadavaliBanner = () => {
+
+  const {
+    isFetchingData: isFetchingBannerMessage,
+    data: bannerData,
+  } = {
+    isFetchingData: false,
```
Comment (gurjit03) [inline src/js/components/Banner/ShabadavaliBanner.tsx:10]: @MySmilingTurban why are we creating this object here with `isFetchingData` property when we already knew it's false ?

File: src/js/components/Banner/ShabadavaliBanner.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    }
+  };
+
+  const lastSeen = sessionStorage.getItem("bannerMessage");
+  const updateLastSeen = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+    e.currentTarget.parentElement?.remove();
+    sessionStorage.setItem("bannerMessage", "seen");
+  };
+
+  if (isFetchingBannerMessage || !bannerData.active || !bannerData.data || lastSeen === 'seen') {
```
Comment (gurjit03) [inline src/js/components/Banner/ShabadavaliBanner.tsx:28]: @MySmilingTurban why we needs to put the `isFetchingBannerMessage` in the condition here? As far as I can see we already have the banner data and not fetching it from network or am I missing something?

**Review by gurjit03 — CHANGES_REQUESTED** [review]

File: src/js/components/Banner/ShabadavaliBanner.tsx (hunk 1 above)
Comment (MySmilingTurban) [reply src/js/components/Banner/ShabadavaliBanner.tsx:22]: Hanji, makes sense
Thanks for pointing this out

File: src/js/components/Banner/ShabadavaliBanner.tsx [hunk 4]
```diff
@@ -2,36 +2,26 @@ import React from 'react';
...
-    }
+  const bannerData = {
+    message: "Help us preserve the Punjabi language! Donate on Dasvandh Network and double your impact with matching!",
+    link: "dvnetwork.org/learnpunjabi",
+    label: "Donate now",
+    type: "3",
   };
 
-  const lastSeen = sessionStorage.getItem("bannerMessage");
+  const lastSeen = sessionStorage.getItem("shbdvliBannerMsg");
```
Comment (gurjit03) [inline src/js/components/Banner/ShabadavaliBanner.tsx:12]: @MySmilingTurban 😅 equally good with adding the complete name `shabdavliBannerMessage`. More clear as we already bannerMessage state.

File: src/js/components/Banner/ShabadavaliBanner.tsx [hunk 5]
```diff
@@ -2,36 +2,26 @@ import React from 'react';
...
   const updateLastSeen = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
     e.currentTarget.parentElement?.remove();
-    sessionStorage.setItem("bannerMessage", "seen");
+    sessionStorage.setItem("shbdvliBannerMsg", "seen");
   };
 
-  if (isFetchingBannerMessage || !bannerData.active || !bannerData.data || lastSeen === 'seen') {
+  if (!bannerData || lastSeen === 'seen') {
```
Comment (gurjit03) [inline src/js/components/Banner/ShabadavaliBanner.tsx:18]: @MySmilingTurban We already have the static bannerData with us, don't see any reason to use this condition also. We can safely remove this check ji.

**Review by gurjit03 — APPROVED** [review]
LGTM !

**Pull request merged**